src/Definition.php

```diff
@@ -6,94 +6,86 @@
-     */
+    /** @var bool */
     protected $overwriteExistingMethods = false;
 
-    /**
-     * Constructor
-     *
-     * @param  null|array $methods
-     */
-    public function __construct($methods = null)
+    public function __construct(?array $methods = null)
     {
         if (is_array($methods)) {
             $this->setMethods($methods);
         }
     }
 
-    /**
-     * Set flag indicating whether or not overwriting existing methods is allowed
-     *
-     * @param mixed $flag
-     * @return \Laminas\Server\Definition
-     */
-    public function setOverwriteExistingMethods($flag)
+    public function setOverwriteExistingMethods(bool $flag): self
     {
-        $this->overwriteExistingMethods = (bool) $flag;
+        $this->overwriteExistingMethods = $flag;
         return $this;
     }
 
     /**
      * Add method to definition
      *
-     * @param  array|\Laminas\Server\Method\Definition $method
-     * @param  null|string $name
-     * @return \Laminas\Server\Definition
-     * @throws \Laminas\Server\Exception\InvalidArgumentException if duplicate or invalid method provided
+     * @param  array|Method\Definition $method
+     * @return $this
+     * @throws InvalidArgumentException if duplicate or invalid method provided
      */
-    public function addMethod($method, $name = null)
+    public function addMethod($method, ?string $name = null): self
     {
         if (is_array($method)) {
             $method = new Method\Definition($method);
         } elseif (! $method instanceof Method\Definition) {
-            throw new Exception\InvalidArgumentException('Invalid method provided');
+            throw new InvalidArgumentException('Invalid method provided');
         }
 
         if (is_numeric($name)) {
             $name = null;
         }
+
         if (null !== $name) {
             $method->setName($name);
         } else {
             $name = $method->getName();
         }
         if (null === $name) {
-            throw new Exception\InvalidArgumentException('No method name provided');
+            throw new InvalidArgumentException('No method name provided');
         }
 
         if (! $this->overwriteExistingMethods && array_key_exists($name, $this->methods)) {
-            throw new Exception\InvalidArgumentException(sprintf('Method by name of "%s" already exists', $name));
+            throw new InvalidArgumentException(sprintf('Method by name of "%s" already exists', $name));
         }
         $this->methods[$name] = $method;
         return $this;
     }
 
-    /**
-     * Add multiple methods
-     *
-     * @param  array $methods Array of \Laminas\Server\Method\Definition objects or arrays
-     * @return \Laminas\Server\Definition
-     */
-    public function addMethods(array $methods)
+    public function addMethods(array $methods): self
     {
         foreach ($methods as $key => $method) {
+            if (is_numeric($key)) {
+                $key = null;
+            }
```

Comment:
`is_numeric` is no need here, as next line is calling `addMethod()` which has `is_numeric` check agains key as a name, make it unneded double check.

Reply:
It is required, because a numeric (integer) value passed to `addMethod()` will throw a TypeError.

Array keys of a numeric string type (e.g. `'100'`) will be cast to an integer internally by PHP. This case is covered by `DefinitionTest::testConstructorNumericKeyWillBeReplacedByMethodName()`.